applications/software-libraries.md

```diff
@@ -0,0 +1,59 @@
+
+This may seem like stating the obvious, but it is essential to allow the application to be checked for any issues in libraries that it depends on.
+
+## Capture your client-side dependencies too
+
+Ensure that any client-side libraries you use are captured in your dependencies.
+
+Again, this sounds obvious but not all client-side frameworks and libraries are necessarily included in your application through bundler or npm.
+
+> As an example: The GOV.UK frontend toolkit includes the GPL-licensed Nomensa accessible media player.  If you link to this in your client-side Javascript you will have to GPL *all* your client-side code.  Does the toolkit's MIT licence cover this?  No.  Is this dependency listed anywhere in the toolkit?  No.  Mind you, IANAL so maybe none of this matters.
+
+## Use a dependency checker
+
+With all your dependencies centrally listed, an automated dependency checker can be used to manage your exposure to
+
+- security vulnerabilities
+- bugs
+- licensing issues
+
+in your dependencies and all of their dependencies.
+
+For Rails projects we use Brakeman and [TBC...]
+
+For Node projects we use [TBC...]
+
+For Java projects we use [TBC...]
+
+## Check licences
+
+Before adding any library to your application, check the licence and copyright that applies to it.
+
+Any commercial licensing will need to be handled by the Commercial team.  Contact Ben Sagar in the first instance, who will be probably know who to start the conversation with in commercial.
+
+Any [copyleft](https://www.gnu.org/licenses/copyleft.en.html) licence conditions actually make it difficult for us to make our code as freely available as we would like so these should be avoided.  Any use of a library with such a licensing condition would need legal consideration.
+
+Generally speaking, if a library uses the [GNU GPL](https://www.gnu.org/licenses/licenses.en.html#GPL) then just don't use it.
+
+## Check repositories
+
+Before adding any library to your application, check where it is held and how it is maintained.
+
+Libraries held in personal repositories with minimal maintenance histories and long outstanding issues might not be the best choice for inclusion in an application.
+
+Where this is the case, check the licence and copyright that apply to the library, as it may be possible to copy it into our own repository.  An increasing number of libraries are using [Copyfree](http://copyfree.org/) licences that facilitate this approach.
+
+If in any doubt, consult with your team's senior developer.  If you *are* your team's senior developer and you are still in any doubt then speak to Ben Sagar, who will probably be able to get some relevant people together to discuss it.
+
+## Create organisational GitHub repositories
+
+If there is a piece of common code that could be a library then create a GitHub repository for it under the EnvironmentAgency organisation.
```

Comment:
@Cruikshanks exactly.  Given we have a private GitLab instance, I am not sure when a new private-facing repo would go onto GitHub vs GitLab, so guidance on this would be useful.